### PR TITLE
Deere :: 2.2 fixes

### PR DESCRIPTION
**Pull request by ronso0**
Major changes:
* unify skin spacing (gaps in between mixer/decks, decks/FX, FX/samplers ....)
* remove exapnding spacer in between sampler left & right blocks
* fix knob labels (qt5 doesn't allow negative margin, that's why labels where cut off)

minor fixes in loop section, Mic/Aux units and AutoDJ control row, skin settings (sampler row selection button right-click)

before
deere-2 2-before

after
deere-2 2-after

### CODE REVIEW

**Comment by uklotzde** [comment]
Thanks for all your work, @ronso0! The UI designs are both appealing and effective :+1:

**Comment by ronso0** [comment]
Thank YOU! without the work of you all my design labor would be...well..pointless. The code would work with ugly skins, but not vice versa ;)

**Comment by uklotzde** [comment]
Have the SVGs already been stripped like for LateNight? The shrink in size is impressive.

**Comment by ronso0** [comment]
Yes, LateNight SVGs are stripped, and IMO it's the only skin where this really was necessary. While vetorizing button icons I started with one SVG, embedded an bitmap to adopt borders, gradients etc. Although I removed the bitmap, this (and other stuff) was still present in that SVG I used for all buttons.

SVGs in Deere & Tango are already quite compact. I'll play around with scour a bit more to check out the advanced options.

File: res/skins/Deere/loop.xml [hunk 1]
```diff
@@ -92,8 +90,10 @@
           </Template>
 
           <!-- Workaround for layout spacing -->
+          <Template src="skin:spacer_h.xml"/>
+
           <Template src="skin:spacer_hx.xml">
-            <SetVariable name="width">8f,</SetVariable>
+            <SetVariable name="width">1f,</SetVariable>
```
Comment (beenisss) [inline res/skins/Deere/loop.xml]: Trailing comma, not sure if this matters.

File: res/skins/Deere/microphone.xml [hunk 2]
```diff
@@ -92,27 +104,32 @@
...
+            <Layout>horizontal</Layout>
+            <Children>
+              <Template src="skin:fx_unit_group_assignment_button.xml">
+                <SetVariable name="EffectUnit">3</SetVariable>
+                <SetVariable name="SourceType">microphone</SetVariable>
+              </Template>
+
+              <!-- Workaround for layout spacing -->
+              <Template src="skin:spacer_hx.xml">
+                <SetVariable name="width">2f,</SetVariable>
```
Comment (beenisss) [inline res/skins/Deere/microphone.xml]: Trailing comma, not sure if this matters.

**Review by benis — APPROVED** [review]
Looking good on Mac.

**Comment by beenisss** [comment]
If you want me to check how the skin displays again, let me know.

**Comment by ronso0** [comment]
I stripped rubbish from SVGs, so please test if this went well for all icons, spinnies etc.

If this worked out, this is ready to merge.

**Comment by ninomp** [comment]
I noticed one minor issue with waveform markers for main cue and loop in/out: they don't resize when you resize the waveforms. This is because these markers are PNGs 119 pixels high. I suggest to remove these 3 images. Other skins don't have images for waveform markers.

**Comment by ronso0** [comment]
yes, good catch.
Icons are okay there, even if they aren't scaled they could be used in conjunction with a line color, but that wouldn't work. Just filed https://bugs.launchpad.net/mixxx/+bug/1794963

For now I'll try to find a suitable glyph for the <Text> node and use the line <Color>

**Comment by beenisss** [comment]
Still looking good to me.

**Comment by beenisss** [comment]
When you think you're done with tweaks etc. feel free to ping me to test again.

**Comment by beenisss** [comment]
Just checked this over again. Looking tight.

**Comment by uklotzde** [comment]
Still LGTM. I'm planning a new snapshot build for RPM Fusion and would like to include all these nice polishing changes.

**Pull request merged**